RecoMET/METPUSubtraction/interface/PFMETAlgorithmMVA.h

```diff
@@ -83,41 +76,29 @@ class PFMETAlgorithmMVA
 
   int    mvaType_;
   bool   hasPhotons_;
- 
-  Float_t pfSumEt_;
-  Float_t pfU_;
-  Float_t pfPhi_;
-  Float_t tkSumEt_;
-  Float_t tkU_;
-  Float_t tkPhi_;
-  Float_t npuSumEt_;
-  Float_t npuU_;
-  Float_t npuPhi_;
-  Float_t puSumEt_;
-  Float_t puMEt_;
-  Float_t puPhi_;
-  Float_t pucSumEt_;
-  Float_t pucU_;
-  Float_t pucPhi_;
-  Float_t jet1Pt_;
-  Float_t jet1Eta_;
-  Float_t jet1Phi_;
-  Float_t jet2Pt_;
-  Float_t jet2Eta_;
-  Float_t jet2Phi_;
-  Float_t numJetsPtGt30_;
-  Float_t numJets_;
-  Float_t numVertices_;
-
-  Float_t* mvaInputU_;
-  Float_t* mvaInputDPhi_;
-  Float_t* mvaInputCovU1_;
-  Float_t* mvaInputCovU2_;
+
+  double dZcut_;
+  float* createFloatVector(std::vector<std::string> variableNames);
```

Comment:
Please use std::unique_ptr<float> as the return type for this method. It protects against memory leaks even in the case of exceptions.